src/nomad_simulations/schema_packages/properties/energies.py

```diff
@@ -20,33 +20,65 @@
 
 import numpy as np
 
-from nomad.metainfo import Quantity
+from nomad.metainfo import Quantity, Section, Context, SubSection
 
 if TYPE_CHECKING:
     from nomad.metainfo import Section, Context
     from nomad.datamodel.datamodel import EntryArchive
     from structlog.stdlib import BoundLogger
 
-from nomad_simulations.schema_packages.physical_property import PhysicalProperty
+from nomad_simulations.schema_packages.physical_property import (
+    PhysicalProperty,
+    PropertyContribution,
+)
 
+##################
+# Abstract classes
+##################
 
-class FermiLevel(PhysicalProperty):
+
+class BaseEnergy(PhysicalProperty):
     """
-    Energy required to add or extract a charge from a material at zero temperature. It can be also defined as the chemical potential at zero temperature.
+    Abstract physical property section describing some energy of a (sub)system.
     """
 
-    # ! implement `iri` and `rank` as part of `m_def = Section()`
-
-    iri = 'http://fairmat-nfdi.eu/taxonomy/FermiLevel'
-
     value = Quantity(
         type=np.float64,
         unit='joule',
         description="""
-        The value of the Fermi level.
+        The value of the energy.
         """,
     )
 
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class EnergyContribution(BaseEnergy, PropertyContribution):
+    """
+    Abstract physical property section linking a property contribution to a contribution
```

Comment:
maybe improve a bit here the description?

Reply:
What do you think of this:
"""
Abstract class for incorporating specific energy contributions to the `TotalEnergy`. The inheritance from `PropertyContribution` allows to link this contribution to a specific component (of class `BaseModelMethod`) of the over `ModelMethod` using the `model_method_ref` quantity. 

For example, for a force field calculation, the `model_method_ref` may point to a particular potential type (e.g., a Lennard-Jones potential between atom types X and Y), while for a DFT calculation, it may point to a particular exchange-correlation functional (e.g., `PBE`). Then, the contribution will be named according to this model component and the `value` quantity will contain the energy contribution from this component evaluated over all relevant atoms or electrons or as a function of them.
"""

Reply:
Just change the phrase of DFT:
 while for a DFT calculation, it may point to a particular exchange-correlation functional (e.g., PBE)

for:
while for a DFT calculation, it may point to a particular term in the electronic interaction terms (e.g., 'XC' for the exchange-correlation term, or 'Hartree' for the Hartree term).